daft/io/writer.py

```diff
@@ -27,23 +37,42 @@ def __init__(
         partition_values: Optional[Table] = None,
         compression: Optional[str] = None,
         io_config: Optional[IOConfig] = None,
-        default_partition_fallback: str = "__HIVE_DEFAULT_PARTITION__",
+        version: Optional[int] = None,
+        resolved_path_and_fs: Optional[Tuple[str, pafs.FileSystem]] = None,
+        default_partition_fallback: Optional[str] = None,
     ):
-        [self.resolved_path], self.fs = _resolve_paths_and_filesystem(root_dir, io_config=io_config)
+        if resolved_path_and_fs is None:
+            [resolved_path], self.fs = _resolve_paths_and_filesystem(root_dir, io_config=io_config)
+        else:
+            resolved_path, self.fs = resolved_path_and_fs
```

Comment:
This is only a nit but: to me this is slightly confusing because it's essentially an indirect way to allow you to use either the default or alternative function to resolve the path and filesystem. What if instead, `FileWriterBase` has a `resolve_paths_and_filesystem` method that it uses to do this, and it can be overwritten by `DeltalakeWriter`?